packages/mobx-state-tree/src/core/node/object-node.ts

```diff
@@ -344,9 +343,8 @@ export class ObjectNode<C, S, T> extends BaseNode<C, S, T> {
     private _snapshotUponDeath?: S
 
     // advantage of using computed for a snapshot is that nicely respects transactions etc.
-    @computed
     get snapshot(): S {
-        return freeze(this.getSnapshot())
+        return this._snapshotComputed.get()
```

Comment:
This is a great PR @urugator!

I'm don't have a great understanding of the MST codebase, so excuse me if my comments are silly.
Could it be of value to use `makeObservable(this, { snapshot: computed });` in the constructor instead? [`makeObservable` seems to be pretty cheap with given annotations](https://github.com/mobxjs/mobx/blob/28f8a11d8b94f1aca2eec4ae9c5f45c5ea2f4362/packages/mobx/src/api/makeObservable.ts#L26).
If not, maybe we should remove the comment above the method?

Reply:
Every time a constructor is called (which is quite often a suppose) it has to:
[create observable object administration](https://github.com/mobxjs/mobx/blob/28f8a11d8b94f1aca2eec4ae9c5f45c5ea2f4362/packages/mobx/src/types/observableobject.ts#L640-L668)
`Reflect.ownKeys` on passed object 
[find closest getter in proto chain](https://github.com/mobxjs/mobx/blob/28f8a11d8b94f1aca2eec4ae9c5f45c5ea2f4362/packages/mobx/src/types/computedannotation.ts#L21-L37)
[define new computed property](https://github.com/mobxjs/mobx/blob/28f8a11d8b94f1aca2eec4ae9c5f45c5ea2f4362/packages/mobx/src/types/observableobject.ts#L439-L491)
VS 
creation of single computed value

Your call :)

> If not, then maybe we should remove the comment above the method?

Remove? or move above the `computed` creation? I left it there as there is similar comment right after near `getSnapshot()`. It seemed better to keep it near the exposed part as that's what consumers see first.

Reply:
Yeah, creation of a single computed value per instance is definitely cheaper, for sure. What I was trying to get at was the the `snapshot` getter will technically not be a computed anymore, right? Maybe I'm nitpicking.

Reply:
Depens how you look at it, the getter is always just a getter that delegates to an underlying computed value (normally exists on ObservableObjectAdministration).
However - and I can be wrong about this as I am not very familar with MST - I think the gist of the comment is to point that the "snapshot" retrieved via this getter will be *computed* - cached, reactive etc.
Obviously you could remove the getter and expose the computed directly, but that would mean to change the API.